torch/nested/_internal/ops.py

```diff
@@ -1586,6 +1586,36 @@ def _nested_get_max_seqlen(func, *args, **kwargs):
     return inp._metadata_cache.get("max_seqlen", None)
 
 
+# If a section of the Nested Tensor is fully masked out we still retain the section with a length of 0
+@register_jagged_func(torch.ops.aten.masked_select.default, "self: jt, mask: any")
+def masked_select_default(func, *args, **kwargs):
+    _, new_kwargs = normalize_function(
+        func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
+    )
+    inp = new_kwargs.pop("input")
+    mask = new_kwargs.pop("mask")
+
+    mask_values = None
+    if inp.ndim > 2:
+        raise RuntimeError("masked_select only support 2-D selections currently")
+    elif inp.shape == mask.shape:
+        mask_values = mask.values()
+    else:
+        raise RuntimeError(
+            f"Mask with shape {mask.shape} is not compatible with input's shape {inp.shape}"
+        )
+    res_values = inp._values.masked_select(mask_values)
+    mask_cumsum = F.pad(mask_values.cumsum(dim=0), (1, 0))
+    res_offsets = mask_cumsum[inp._offsets]
+
+    return NestedTensor(
+        values=res_values,
+        offsets=res_offsets,
+        _metadata_cache=inp._metadata_cache,
+        _ragged_idx=inp._ragged_idx,
+    )
```

Comment:
nit: could we use the `extract_kwargs(indices)` utility here, then replace the offsets